tensilelite/Tensile/SolutionStructs.py

```diff
@@ -2668,28 +2683,30 @@ def assignDerivedParameters(state):
       state["UnrollMajorLDSMetadata"] = state["TransposeLDSMetadata"] and (not state["ProblemType"]["TLUMetadata"])
 
     auto_LdsBlockSizePerPadA_for_mix = 0
+    tmpBpe = state["ProblemType"]["DataTypeA"].numBytes() if state["ConvertAfterDS"] else state["ProblemType"]["DataType"].numBytes()
     if state["LdsBlockSizePerPadA"] == -1:
       auto_LdsBlockSizePerPadA_for_mix = 1
       if state["UnrollMajorLDSA"]:
-        state["LdsBlockSizePerPadA"] = roundUpToNearestMultiple(state["_DepthUA"] * state["ProblemType"]["DataType"].numBytes(), 128)
-        if state["_DepthUA"] * state["ProblemType"]["DataType"].numBytes() * state["VectorWidthA"] > 128:
-          state["LdsBlockSizePerPadA"] = roundUpToNearestMultiple(state["_DepthUA"] * state["ProblemType"]["DataType"].numBytes() * state["VectorWidthA"], 128)
+        state["LdsBlockSizePerPadA"] = roundUpToNearestMultiple(tmpBpe, 128)
+        if state["_DepthUA"] * tmpBpe * state["VectorWidthA"] > 128:
+          state["LdsBlockSizePerPadA"] = roundUpToNearestMultiple(state["_DepthUA"] * tmpBpe * state["VectorWidthA"], 128)
       else:
         if state["MatrixInstB"] == 1 and state["MatrixInstM"] == 16:
-          state["LdsBlockSizePerPadA"] = state["MacroTile0"] * state["ProblemType"]["DataType"].numBytes() * state["LocalReadVectorWidth"]
+          state["LdsBlockSizePerPadA"] = state["MacroTile0"] * tmpBpe * state["LocalReadVectorWidth"]
         else:
           state["LdsBlockSizePerPadA"] = 0
 
     auto_LdsBlockSizePerPadB_for_mix = 0
+    tmpBpe = state["ProblemType"]["DataTypeB"].numBytes() if state["ConvertAfterDS"] else state["ProblemType"]["DataType"].numBytes()
     if state["LdsBlockSizePerPadB"] == -1:
       auto_LdsBlockSizePerPadB_for_mix = 1
       if state["UnrollMajorLDSB"]:
-        state["LdsBlockSizePerPadB"] = roundUpToNearestMultiple(state["_DepthUB"] * state["ProblemType"]["DataType"].numBytes(), 128)
-        if state["_DepthUB"] * state["ProblemType"]["DataType"].numBytes() * state["VectorWidthB"] > 128:
-          state["LdsBlockSizePerPadB"] = roundUpToNearestMultiple(state["_DepthUB"] * state["ProblemType"]["DataType"].numBytes() * state["VectorWidthB"], 128)
+        state["LdsBlockSizePerPadB"] = roundUpToNearestMultiple(tmpBpe, 128)
```

Comment:
Should it be state["_DepthUB"] * tmpBpe?

Reply:
update